GHzDACs/ghz_fpga_server.py

```diff
@@ -1644,18 +1659,66 @@ def dac_run_sram(self, c, data, loop=False, blockDelay=0):
         parameters specifies the number of microseconds to delay
         for a multiblock sequence.
         """
-        if not len(data):
-            return
-        
-        if loop:
-            # make sure data is at least 20 words long by repeating it
-            data *= (20-1)/len(data) + 1
-        else:
-            # Set data at least 20 words long by repeating first value
-            data = list(data) + [data[0]] * (20-len(data))
-        
+        if len(data) < 20:
+            raise ValueError("Cannot play less than 20 ns of data.")
+
         dev = self.selectedDAC(c)
         yield dev.runSram(data, loop, blockDelay)
+
+    @setting(2081, 'DAC Write SRAM', data='*w')
+    def dac_write_sram(self, c, data):
```

Comment:
There is no other function that _just_ writes SRAM (i.e. outside of `run_sequence` and without playing it). This was mainly for debugging purposes--we thought we might use it in a clever manner for the board calibrations but ended up not using it. I will add better documentation.